src/ReactART.js

```diff
@@ -329,315 +88,446 @@ const NodeMixin = {
+    // actually resolve to because ART may render canvas, vml or svg tags here.
+    // We only allow a subset of properties since others might conflict with
+    // ART's properties.
+    const props = this.props;
+
+    // TODO: ART's Canvas Mode overrides surface title and cursor
+    const Tag = Mode.Surface.tagName;
 
-// Declarative fill type objects - API design not finalized
+    return (
+      <Tag
+        ref={ref => this._tagRef = ref}
+        accessKey={props.accessKey}
+        className={props.className}
+        draggable={props.draggable}
+        role={props.role}
+        style={props.style}
+        tabIndex={props.tabIndex}
+        title={props.title}
+      />
+    );
+  }
+}
+
+/** Declarative fill-type objects; API design not finalized */
 
 const slice = Array.prototype.slice;
 
-function LinearGradient(stops, x1, y1, x2, y2) {
-  this.args = slice.call(arguments);
+class LinearGradient {
+  constructor(stops, x1, y1, x2, y2) {
+    this._args = slice.call(arguments);
+  }
+
+  applyFill(node) {
+    node.fillLinear.apply(node, this._args);
+  }
 }
 
-LinearGradient.prototype.applyFill = function(node) {
-  node.fillLinear.apply(node, this.args);
-};
+class RadialGradient {
+  constructor(stops, fx, fy, rx, ry, cx, cy) {
+    this._args = slice.call(arguments);
+  }
 
-function RadialGradient(stops, fx, fy, rx, ry, cx, cy) {
-  this.args = slice.call(arguments);
+  applyFill(node) {
+    node.fillRadial.apply(node, this.args);
+  }
 }
 
-RadialGradient.prototype.applyFill = function(node) {
-  node.fillRadial.apply(node, this.args);
-};
+class Pattern {
+  constructor(url, width, height, left, top) {
+    this._args = slice.call(arguments);
+  }
 
-function Pattern(url, width, height, left, top) {
-  this.args = slice.call(arguments);
+  applyFill(node) {
+    node.fillImage.apply(node, this.args);
+  }
 }
 
-Pattern.prototype.applyFill = function(node) {
-  node.fillImage.apply(node, this.args);
-};
+/** RENDERER */
+
+const ARTRenderer = ReactFiberReconciler({
+  appendChild(parentInstance, child) {
+    child.instance.inject(parentInstance.instance);
+  },
+
+  appendInitialChild(parentInstance, child) {
+    child.instance.inject(parentInstance.instance);
+  },
+
+  commitTextUpdate(textInstance, oldText, newText) {
+    invariant(false, 'ReactART does not support text children');
+  },
+
+  commitUpdate(instance, oldProps, newProps) {
+    instance.applyProps(newProps, oldProps);
+  },
+
+  createInstance(type, props, internalInstanceHandle) {
+    let instance;
+
+    switch (type) {
+      case TYPES.CLIPPING_RECTANGLE:
+        instance = new ClippingRectangle(props);
+        break;
+      case TYPES.GROUP:
+        instance = new Group(props);
+        break;
+      case TYPES.SHAPE:
+        instance = new Shape(props);
+        break;
+      case TYPES.TEXT:
+        instance = new Text(props);
+        break;
+    }
+
+    return instance;
+  },
+
+  createTextInstance(text, internalInstanceHandle) {
```

Comment:
I tried that as well, but it didn't work for reasons I explained above. (Maybe one of us is misunderstanding each other?) If it would be easier to chat on Hangouts I could. 😄